.github/skills/agent-deck-contributor/scripts/self-check.sh

```
@@ -0,0 +1,275 @@
+
+  # 6. revert-check — your changed tests must FAIL on the base without your fix.
+  #    (Mirrors the correctness lens: revert the core hunks, re-run the tests.)
+  if [ -n "$GO_PROD_CHANGED" ] && [ -n "$GO_TESTS_CHANGED" ] && [ "${SKIP_REVERT_CHECK:-0}" != 1 ]; then
+    WT=$(mktemp -d)/wt
+    if git worktree add -q "$WT" "$MB" 2>/dev/null; then
+      RC_OK=1
+      for f in $GO_TESTS_CHANGED; do
+        mkdir -p "$WT/$(dirname "$f")"; cp "$f" "$WT/$f"
+      done
+      RC_PKGS=$(printf '%s\n' "$GO_TESTS_CHANGED" | xargs -n1 dirname | sort -u | sed 's|^|./|' | tr '\n' ' ')
+      if OUT=$( cd "$WT" && sandboxed go test $RC_PKGS 2>&1 ); then
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)